### PR TITLE
Deploy SQL Private Endpoint and Private DNS

### PR DESCRIPTION
**Pull request by nbowes24**

### CODE REVIEW

File: templates/environment.template.json [hunk 1]
```diff
@@ -109,6 +109,14 @@
         "description": "The number of shared worker app subnets to provision."
       }
     },
+    "sharedSqlServerSubnetCount": {
```
Comment (jack-cloud-platform) [inline templates/environment.template.json:112]: Could this be a more generic private endpoint subnet as we could have multiple private endpoints deployed to the same subnet? https://docs.microsoft.com/en-us/azure/private-link/private-link-faq#can-i-create-multiple-private-endpoints-in-same-vnet--can-they-connect-to-different-services--

Comment (nbowes24) [reply templates/environment.template.json:112]: Good idea although the idea is in a future story we'll associate an NSG with this subnet and have ASG rules specific to SQL access

Comment (jack-cloud-platform) [reply templates/environment.template.json:112]: Ah good point, as discussed I think maxValue for the subnetCount of 1 is likely suitable.

File: templates/environment.template.json [hunk 2]
```diff
@@ -192,14 +214,17 @@
...
     "virtualNetworkName": "[concat(variables('resourceNamePrefix'),'-vnet')]",
     "subnetServiceEndpointList": [
       "Microsoft.Web",
       "Microsoft.Sql",
       "Microsoft.Storage",
       "Microsoft.AzureCosmosDB",
       "Microsoft.KeyVault"
-    ]
+    ],
+    "envrionmentExternalSqlServerPrivateEndpoints": "[parameters('externalSqlServerPrivateEndpoints')[toUpper(variables('environmentName'))]]"
```
Comment (jack-cloud-platform) [inline templates/environment.template.json:227]: Typo envrioment

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**